### PR TITLE
test.py: improve logging

### PR DESCRIPTION
**Pull request by kbr-scylla**
Make it easy to see which clusters are operated on by which tests in which build modes and so on.
Add some additional logs.

These improvements would have saved me a lot of debugging time if I had them last week and we would have https://github.com/scylladb/scylladb/pull/12482 much faster.

### CODE REVIEW

**Comment by kbr-scylla** [comment]
Sneak peek:
```
19:20:43.814 INFO> [dev/topology.test_topology.1] Created new cluster 54820316-904a-11ed-829f-f4d108a9e4a3
19:20:43.814 INFO> [dev/topology.test_topology.1] Cluster 54820316-904a-11ed-829f-f4d108a9e4a3 waiting for new IP...
19:20:43.830 INFO> [dev/topology.test_topology.2] Setting up dev/topology.test_topology.2::test_stop_server_add_column
19:20:43.830 INFO> [dev/topology.test_topology.2] Current cluster 49ef6e02-904a-11ed-829f-f4d108a9e4a3 is dirty after last test, stopping...
19:20:43.830 INFO> [dev/topology.test_topology.2] Cluster ScyllaCluster(name: 49ef6e02-904a-11ed-829f-f4d108a9e4a3, running: ScyllaServer(2, 127.171.184.2, c9346d82-a773-4234-8074-199ae64830e0), ScyllaServer(5, 127.171.184.5, 0e448c6f-bf5e-4ff4-83c6-525c
6db3483e), ScyllaServer(8, 127.171.184.8, 6b82ed0a-b481-4be1-b772-de11f20aa847), ScyllaServer(12, 127.171.184.12, c6f86792-f197-4fed-8bd7-91222199ab3b), stopped: ) stopping
19:20:43.830 INFO> [dev/topology.test_topology.2] stopping ScyllaServer(2, 127.171.184.2, c9346d82-a773-4234-8074-199ae64830e0) in scylla-2
19:20:43.919 INFO> [dev/topology.test_topology.2] stopping ScyllaServer(5, 127.171.184.5, 0e448c6f-bf5e-4ff4-83c6-525c6db3483e) in scylla-5
19:20:44.001 INFO> [dev/topology.test_topology.2] stopping ScyllaServer(8, 127.171.184.8, 6b82ed0a-b481-4be1-b772-de11f20aa847) in scylla-8
19:20:44.102 INFO> [dev/topology.test_topology.2] stopping ScyllaServer(12, 127.171.184.12, c6f86792-f197-4fed-8bd7-91222199ab3b) in scylla-12
19:20:44.128 INFO> [dev/topology.test_topology.2] stopped ScyllaServer(2, 127.171.184.2, c9346d82-a773-4234-8074-199ae64830e0) in scylla-2
19:20:44.128 INFO> [dev/topology.test_topology.2] stopped ScyllaServer(5, 127.171.184.5, 0e448c6f-bf5e-4ff4-83c6-525c6db3483e) in scylla-5
19:20:44.129 INFO> [dev/topology.test_topology.3] Finished test dev/topology.test_topology.3::test_add_server_add_column, cluster: ScyllaCluster(name: 49f78dee-904a-11ed-829f-f4d108a9e4a3, running: ScyllaServer(3, 127.171.184.3, f9556c64-de62-4785-ae42-3
30a3db0806f), ScyllaServer(6, 127.171.184.6, 0534f499-c0c6-491f-9b57-7cc47368c755), ScyllaServer(9, 127.171.184.9, 26f1e260-22df-47a2-8687-fa66dda9002c), ScyllaServer(11, 127.171.184.11, 4f076f83-b197-4bb1-a24f-994088d8a5dc), stopped: )
19:20:44.133 INFO>  [09/Jan/2023:18:20:44 +0000] "GET /cluster/after-test HTTP/1.1" 200 155 "-" "Python/3.11 aiohttp/3.8.3"
19:20:44.134 INFO> [dev/topology.test_topology.2] stopped ScyllaServer(8, 127.171.184.8, 6b82ed0a-b481-4be1-b772-de11f20aa847) in scylla-8

```

notice the `[dev/topology.test_topology.2]`.

**Review by avikivity — APPROVED** [review]

**Review by denesb — APPROVED** [review]

**Comment by denesb** [comment]
Conflicts, please rebase.

**Review by nyh — APPROVED** [review]

File: test.py [hunk 1]
```diff
@@ -877,7 +877,7 @@ async def run(self, options: argparse.Namespace) -> Test:
         self._prepare_pytest_params(options)
 
         test_path = os.path.join(self.suite.options.tmpdir, self.mode)
-        async with get_cluster_manager(self.uname, self.suite.clusters, test_path) as manager:
+        async with get_cluster_manager(self.mode + '/' + self.uname, self.suite.clusters, test_path) as manager:
```
[the remark this answers is not in the document]
Comment (avikivity) [reply test.py:886]: It's the first run of dev/topology.test_topology, so the name makes sense IMO.

**Comment by kbr-scylla** [comment]
Rebased

**Comment by denesb** [comment]
> CI state **`FAILURE`** - https://jenkins.scylladb.com/job/releng/job/Scylla-CI/3766/

[Sanity Tests / ttl_test.TestDistributedTTL.test_ttl_is_respected_on_repair](https://jenkins.scylladb.com/job/releng/job/Scylla-CI/3766/testReport/junit/ttl_test/TestDistributedTTL/Sanity_Tests___test_ttl_is_respected_on_repair/)

This is a known flaky test: https://github.com/scylladb/scylladb/issues/11947.
Re-kicked CI.

File: test.py [hunk 2]
```diff
@@ -836,10 +840,12 @@ async def run(self, options: argparse.Namespace) -> Test:
...
         self._prepare_pytest_params(options)
 
-        async with self.suite.clusters.instance() as cluster:
+        loggerPrefix = self.mode + '/' + self.uname
+        logger = LogPrefixAdapter(logging.getLogger(loggerPrefix), {'prefix': loggerPrefix})
+        async with self.suite.clusters.instance(logger) as cluster:
             try:
                 cluster.before_test(self.uname)
-                logging.info("Leasing Scylla cluster %s for test %s", cluster, self.uname)
+                logger.info("Leasing Scylla cluster %s for test %s", cluster, self.uname)
```
Comment (kostja) [inline test.py:848]: I don't see how it works when we're reusing an existing cluster. The args are only passed to the build function, the adapter apparently is not reset when taking an exsiting cluster.

File: test/pylib/scylla_cluster.py [hunk 3]
```diff
@@ -763,6 +773,11 @@ def update_config(self, server_id: ServerNum, key: str, value: object) -> Action
         self.servers[server_id].update_config(key, value)
         return ScyllaCluster.ActionReturn(success=True)
 
+    def setLogger(self, logger: logging.LoggerAdapter):
```
Comment (kostja) [inline test/pylib/scylla_cluster.py:779]: Please explain why you need this functio and how it's used in a comment.

**Review by kostja — APPROVED** [review]

**Comment by kostja** [comment]
Asked for a comment against setLogger().

**Comment by kbr-scylla** [comment]
Wrote the comments

**Comment by kbr-scylla** [comment]
@denesb would you merge please?

**Comment by denesb** [comment]
> @denesb would you merge please?

Queued.

**Pull request closed without merging**